In [5]:
import numpy       as np
import pandas      as pd
import scipy.stats as stats
from scipy.stats             import ttest_1samp,ttest_ind, wilcoxon
from statsmodels.stats.power import ttest_power
import matplotlib.pyplot     as     plt

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


## Q1- Q3 Description-  The data collected by an Advertisement agency has revealed that out of 2800 visitors, 56 visitors clicked on 1 Advertisement, 30 clicked on 2 advertisements and 14 clicked on 3 advertisements and the remaining did not click on any advertisement.

Calculate

### Q1 The probability that a visitor to the website will not click on any advertisement.


A) The probability that a visitor to the website will not click on any advertisement.

Here 56 visitors click on 1 advertisement

Here 30 visitors click on 2 advertisements

Here 14 visitors click on 3 advertisements

Hence, the total number of customers clicking on any advertisement is 100

In [9]:
FE1 = 2800 - 100
TE  = 2800
PE1 = round(FE1 / TE,4)
print('a. The probability that a visitor to the website will not click on any advertisement is %1.4f' % PE1)

a. The probability that a visitor to the website will not click on any advertisement is 0.9643


### Q2 The probability that a visitor to the website will click on an advertisement.

In [10]:
FE2 = 100 # 56 + 30 + 14
TE  = 2800
PE2 = round(FE2 / TE,4)
print('b. The probability that a visitor to the website will click on an advertisement is %1.4f' % PE2)

b. The probability that a visitor to the website will click on an advertisement is 0.0357


## Q3 The probability that a visitor to the website will click on more than one advertisement.

In [11]:
FE3 = 44 # = 30 + 14
TE  = 2800
PE3 = round(FE3 / TE,4) 
print('c. The probability that a visitor to the website will click on more than one advertisement. is %1.4f' % PE3)

c. The probability that a visitor to the website will click on more than one advertisement. is 0.0157


## Q4 -5 Description - At a popular company service center, a total of 100 complaints were received. 80 customers complained about late delivery of the items and 60 complained about poor product quality.

## Q4 Calculate the probability that a customer complaint will be about both product quality and late delivery.


In [12]:
FE4 = 40 # = (80 + 60) - 100 
TE  = 100
PE4 = round(FE4 / TE,4) 
print('a.Probability that a customer complaint will\n\
       be about both product quality and late delivery. is %1.4f' % PE4)

a.Probability that a customer complaint will
       be about both product quality and late delivery. is 0.4000


## Q5 - What is the probability that a complaint will be only about late delivery?

In [13]:
# FE5 is the complaints about poor quality
FE5 = 60
TE  = 100
PE5 = round(FE5 / TE,4) 
PE6 = 1 - PE5
# So,1 - PE5 is the Probability that a customer complaint will be about
#        both product quality and late delivery 
print('b.probability that a complaint will be \n\
         only about late delivery. is %1.4f' % PE6)

b.probability that a complaint will be 
         only about late delivery. is 0.4000


## Q6 You are given the daily sugar intake of 11 diabetic patients in the following Python code. Is there any evidence to the claim that the average daily sugar intake of the diabetic patients is 7600 mg.

Hint: Use t test

In [0]:
# daily intake of Sugar in milligrams for 11 diabetic patients
import numpy as np
daily_intake = np.array([5560, 5770, 7640, 5180, 5690, 6435, 6803, 7689, 6876, 8213, 8765])

In [6]:
print('Mean is %2.1f Sd is %2.1f' % (daily_intake.mean(),np.std(daily_intake,ddof = 1)))

Mean is 6783.7 Sd is 1180.9


In [0]:
t_statistic, p_value = ttest_1samp(daily_intake, 7600)

In [8]:
print(t_statistic, p_value)

-2.2925841964555027 0.04481624562570511


In this example, p value is 0.044 and it is less than 5% level of significance

So the statistical decision is to reject the null hypothesis at 5% level of significance.

So there is no sufficient evidence to prove that there is no evidence to the claim
that the average daily sugar intake of the diabetic patients is 7600 mg.

## Q7-Q8 Description Write the null and alternative hypothesis from the following hypopthesis description:
## Q7Average annual salary of Data Scientists is different for those having Ph.D in Statistics and those who do not.
* Let $\mu_{PhD}$ be the average annual salary of a Data scientist with Ph.D in Statistics.
* Let $\mu_{NoPhD}$ be the average annual salary of a Data scientist without Ph.D in Statistics.

## Answer - 
* Null hypothesis:        $H_0$: $\mu_{PhD}$ =    $\mu_{NoPhD}$ 
* Alternative hypothesis: $H_A$: $\mu_{PhD}$ $\neq$ $\mu_{NoPhD}$ 

Since the rejection region is on either side of the distribution, it will be a **two-tailed** test.

## Q8 Average annual salary of Data Scientists is more for those having Ph.D in Statistics and those who do not.

## Answer -
* Null hypothesis:        $H_0$: $\mu_{PhD}$ $\leq$   $\mu_{NoPhD}$ 
* Alternative hypothesis: $H_A$: $\mu_{PhD}$ >        $\mu_{NoPhD}$ 

Since the rejection region is on the right side of the distribution, it will be a one-tailed test.

## Q9 - Compare the following two unrelated samples. Data was collected on the weight of women and men enrolled in a weight reduction program. At 𝛼 α = 0.05, test whether the weight of these two samples is different.

In [0]:
Weight_Female       =  [ 53.8, 54.4, 51.2, 52.5, 61.0, 50.6, 51.6, 70.0]
Weight_Male         =  [ 72.5, 80.3, 71.3, 67.7, 66.2, 73.4, 61.3, 76.8]

NULL HYPOTHSIS MALE AND FEMALE WEIGHT IS EQUAL.

ALTERNATE HYPOTHESIS MALE AND FEMALE WEIGHT IS DIFFERENT.

OUR CONDITION IS TO PROVE THAT WEIGHT OF THE MALE AND FEMALE ARE NOT EQUAL SO,WE REQUIRE ALTERNATE HYPOTHESIS.

In [15]:
t_statistic, p_value  =  stats.ttest_ind(Weight_Female,Weight_Male)
print('P Value %1.3f' % p_value)  

P Value 0.000


In this example, p value is 0.000 and it is LESS than 5% level of significance

So the statistical decision is to REJECT the null hypothesis at 5% level of significance.

So there is sufficient evidence to reject the null hypothesis that the weight of these two samples is different.

## Q10 - A Cable service provider company is interested in checking whether or not the customer churn depends on customer segment. Use 5% as level of significance



| Customer Segment | Churned | Retained |
|  -------- | ----- | ---- |
| S1 | 15 | 142 |
| S2 | 24 | 400 |
| S3 | 30 | 389 |


In [16]:
import numpy       as np
import scipy.stats as stats

churn_array = np.array([[15,142],[24, 400],[30, 389]])
chi_sq_Stat, p_value, deg_freedom, exp_freq = stats.chi2_contingency(churn_array)

print('Chi-square statistic %3.5f P value %1.6f Degrees of freedom %d' %(chi_sq_Stat, p_value,deg_freedom))

Chi-square statistic 2.77998 P value 0.249078 Degrees of freedom 2


In this example, p value is 0.249078 and > 0.05 so we wont reject the null hypothesis.

## End